tests/unit/src/test/scala/tests/decorations/SyntheticDecorationsLspSuite.scala

```diff
@@ -286,4 +286,45 @@ class SyntheticDecorationsLspSuite extends BaseLspSuite("implicits") {
       )
     } yield ()
   }
+
+  test("readonly-files") {
+    for {
+      _ <- server.initialize(
+        s"""|/metals.json
+            |{
+            |  "a": {}
+            |}
+            |
+            |/standalone/Main.scala
```

Comment:
It was easier to test out this issue with standalone files, because readonly files need to be opened from a build target to be associated to a compiler instance.